receiver/statsdreceiver/reporter.go

```diff
@@ -30,15 +40,89 @@ func newReporter(set receiver.CreateSettings) (transport.Reporter, error) {
 	if err != nil {
 		return nil, err
 	}
-	return &reporter{
+
+	r := &reporter{
 		logger:        set.Logger,
 		sugaredLogger: set.Logger.Sugar(),
 		obsrecv:       obsrecv,
-	}, nil
+		staticAttrs: []attribute.KeyValue{
+			attribute.String("receiver", set.ID.String()),
+		},
+	}
+
+	// See https://github.com/open-telemetry/opentelemetry-collector/blob/241334609fc47927b4a8533dfca28e0f65dad9fe/receiver/receiverhelper/obsreport.go#L104
+	// for the metric naming conventions
+
+	r.acceptedMetricPoints, err = metadata.Meter(set.TelemetrySettings).Int64Counter(
+		"receiver/accepted_metric_points",
```

Comment:
Does this receiver not use receiverhelper? If it is it should be producing this metric already.

Reply:
I'm with @TylerHelmuth here. I've read the PR description but still don't understand why we cannot use `receiverhelper.NewObsReport` as other receivers. Excessive tracing is a concern for any receiver, so collector's tracing is expected to be sampled.

Reply:
> Does this receiver not use receiverhelper? If it is it should be producing this metric already.

It does not use `receiverhelper`.  It previously used the `obsreport` package, but that was dead code as mentioned in the issue https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/24278#issuecomment-1636473212

Reply:
> don't understand why we cannot use receiverhelper.NewObsReport as other receivers. 

It's messy to use `obsreport` because the asynchronous nature of this receiver (the flush happening in the background) does not align with the `StartMetricsOp()` / `EndMetricsOp()` pattern of obsreport.  This is likely why the obsreport integration was abandoned in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/1670 when aggregation / async flush was implemented in this receiver.

> Excessive tracing is a concern for any receiver, so collector's tracing is expected to be sampled.

I'm more concerned with the performance overhead of the tracing instrumentation code than I am the volume of traces.  Since submitting this PR we actually had to rewrite this instrumentation because even the overhead of incrementing the metric counter synchronously was causing statsd packet loss.

You can see in the attached flame graph that simply calling `metric.(*int64Inst).Add()` is significant overhead.

![image](https://github.com/open-telemetry/opentelemetry-collector-contrib/assets/381302/4815fbe4-dbcb-41d4-a4b5-35be6f8354a6)